server/src/main/java/hu/flowacademy/epsilon/sport_event_organizer/email/MailService.java

```diff
@@ -0,0 +1,178 @@
+        props.put("mail.smtp.auth", "true");
+        props.put("mail.smtp.host", "smtp.gmail.com");
+        props.put("mail.smtp.port", "587");
+    }
+
+    Session session = Session.getInstance(props,
+            new javax.mail.Authenticator() {
+                protected PasswordAuthentication getPasswordAuthentication() {
+                    return new PasswordAuthentication(username, password);
+                }
+            });
+
+
+    @Async
+    public void sendMailAddUserToTeamMember(User user, Team team) {
+
+        try {
+            Message message = new MimeMessage(session);
+            message.setFrom(new InternetAddress("projektmunkasports@gmail.com"));
+            message.setRecipients(Message.RecipientType.TO,
+                    InternetAddress.parse(user.getEmail()));
+            message.setSubject(team.getName() + " Team Invitation");
+            message.setText("Hello " + user.getGoogleName() + ",\n  Team leader invited you to " + team.getCompany() + "'s " + team.getName() + " team. Check out our page: " + homePageUrl);
+            Transport.send(message);
+        } catch (MessagingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Async
+    public void sendMailDeleteUserFromTeam(User user, Team team) {
+
+        try {
+            Message message = new MimeMessage(session);
+            message.setFrom(new InternetAddress("projektmunkasports@gmail.com"));
+            message.setRecipients(Message.RecipientType.TO,
+                    InternetAddress.parse(user.getEmail()));
+            message.setSubject(team.getName() + " Team delete");
+            message.setText("Hello " + user.getGoogleName() + ",\n  Team leader deleted you from " + team.getCompany() + "'s " + team.getName() + " team. Check out our page: " + homePageUrl);
+            Transport.send(message);
+        } catch (MessagingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Async
+    public void sendMailAddUserToTeamLeader(User user, Team team) {
+
+        try {
+            Message message = new MimeMessage(session);
+            message.setFrom(new InternetAddress("projektmunkasports@gmail.com"));
+            message.setRecipients(Message.RecipientType.TO,
+                    InternetAddress.parse(user.getEmail()));
+            message.setSubject(team.getName() + "Team Promotion");
+            message.setText("Hello " + user.getGoogleName() + ",\n  Team leader promote you to leader in" + team.getCompany() + "'s " + team.getName() + " team. Check out our page: " + homePageUrl);
+            Transport.send(message);
+        } catch (MessagingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Async
+    public void sendMailOrganizersToApplieTeam(Team team, Cup cup) {
+        User[] organizersArr = (User[]) cup.getOrganizers().toArray();
+
+
+        for (int i = 0; i < organizersArr.length; i++) {
```

Comment:
Miért inkrementális for? Nem lehetne inkább a `cup.getOrganizers()`-t használni egy foreach-csel?